lib/jekyll-sitemap.rb

```diff
@@ -50,7 +50,7 @@ def write
 
     def sitemap_content
       site_map = PageWithoutAFile.new(@site, File.dirname(__FILE__), "", "sitemap.xml")
-      site_map.content = File.read(source_path)
+      site_map.content = File.read(source_path).strip.gsub(/\s+/,' ').gsub(/[>}]\s+[<{]/){|s| s.gsub(/\s+/,'')}
```

Comment:
Sooooo many gsubs. Let's pare this down to 1 if possible – each call creates lots of unnecessary objects in memory and is very slow. Or, make the destructive. :)

https://www.igvita.com/2008/07/08/6-optimization-tips-for-ruby-mri/